spyder/utils/tests/test_get_words.py

```diff
@@ -0,0 +1,127 @@
+                      'com', 'def', 'for', 'gmail', 'home', 'i', 'implemented', 'import',
+                      'in', 'int', 'is', 'iterate_', 'jmg', 'neighbours', 'new', 'np', 'null',
+                      'numpy', 'over', 'print', 'range', 'ravel', 'return', 'rules', 'shape',
+                      'stay', 'sure', 'survive', 'utn', 'values', 'zeros']
+    assert sorted(words) == sorted(expected_words)
+
+
+def test_get_words_java():
+    """Test for get word from java file syntax"""
+
+    path = dirname(dirname(abspath(__file__)))
+    f_in = path+"/tests/data/example.java"
+    words = get_words_file(f_in)
+    expected_words = ['Compilation', 'Execution', 'Hello', 'HelloWorld', 'Prints', 'String',
+                      'System', 'World', 'args', 'class', 'java', 'javac', 'main', 'out',
+                      'println', 'public', 'static', 'terminal', 'the', 'to', 'void', 'window']
+    assert sorted(words) == sorted(expected_words)
+
+
+def test_get_words_cplusplus():
+    """Test for get word from C++ file syntax"""
+
+    path = dirname(dirname(abspath(__file__)))
+    f_in = path+"/tests/data/example.cpp"
+    words = get_words_file(f_in)
+    expected_words = ['Consider', 'Create', 'Implement', 'Obj', 'ObjContainer',
+                      'Postfix', 'Prefix', 'Return', 'SmartPointer', 'Static',
+                      'Zero', 'a', 'above', 'access', 'actual', 'add', 'an', 'back',
+                      'bool', 'call', 'class', 'const', 'container', 'cout', 'definitions',
+                      'do', 'end', 'endl', 'f', 'false', 'for', 'friend', 'g', 'i', 'if',
+                      'implement', 'include', 'index', 'indicates', 'int', 'iostream', 'iterator',
+                      'j', 'list', 'main', 'member', 'method', 'namespace', 'o', 'obj', 'objc',
+                      'oc', 'of', 'operator', 'overload', 'pointer', 'public', 'push', 'return',
+                      's', 'size', 'smart', 'sp', 'standard', 'static', 'std', 'sz', 'the',
+                      'to', 'true', 'using', 'value', 'vector', 'version', 'void', 'while']
+    assert sorted(words) == sorted(expected_words)
+
+
+def test_get_words_markdown():
+    """Test for get word from markdown file syntax """
+    path = dirname(dirname(abspath(__file__)))
+    f_in = path+"/tests/data/example.md"
+    words = get_words_file(f_in)
+    expected_words = ['A', 'Blockquote', 'Bold', 'Heading', 'Horizontal', 'Image', 'Inline',
+                      'Italic', 'Link', 'List', 'One', 'Rule', 'Three', 'Two', 'a', 'after',
+                      'b', 'backticks', 'blank', 'block', 'code', 'com', 'http', 'indent',
+                      'jpg', 'line', 'or', 'org', 'paragraph', 'png', 'print', 'spaces',
+                      'url', 'with']
+    assert sorted(words) == sorted(expected_words)
+
+
+def test_get_words_c():
+    """Test for get word from C file syntax"""
+
+    path = dirname(dirname(abspath(__file__)))
+    f_in = path+"/tests/data/example.c"
```

Comment:
You can also use py.test's `tmpdir` fixture, described at http://doc.pytest.org/en/latest/tmpdir.html